libs/core/src/graphql/get-fragment-names.ts

```diff
@@ -0,0 +1,17 @@
+import { DocumentNode, FragmentDefinitionNode } from 'graphql';
+
+const getFragmentNames = (fragment: DocumentNode) =>
+  fragment.definitions.filter(def =>
+    def.kind === 'FragmentDefinition'
+  ).map(def =>
+    (def as FragmentDefinitionNode).name.value
+  )
+
+/**
+ * Builds a list of fragment names present inside the specified GraphQL document nodes.
+ * Returns an empty array if no fragments have been defined or if null is passed.
+ * @param fragments The created fragments.
+ */
+export const daffGetFragmentNames = (...fragments: DocumentNode[]): string[] =>
```

Comment:
Will this ever be used outside of the `daffBuildFragmentNameSpread` function? I can't think of when it would be, and if that's the case, I don't think it should be a part of the public api. I think it should be moved into the same file as `daffBuildFragmentNameSpread`

Reply:
lgtm otherwise

Reply:
I can't think of any use case at the moment. I feel like getting the names of the fragments present in a document definition is a useful utility function to have but we can always easily export it again if we need it. YAGNI for now.

Reply:
changed

Reply:
Probably move the tests too, right? Save time later if it needs moved out

Reply:
nah we don't need to test a function that isn't exported. It'll be part of this PR's diff if we really need it.